src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java

```diff
@@ -515,7 +516,11 @@ public final Response getUserProgressInformation(@Context final HttpServletReque
             }
             userOfInterestSummary = userManager.convertToUserSummaryObject(userOfInterestFull);
 
-            if (associationManager.hasPermission(user, userOfInterestSummary)) {
+            // If user is fetching their own progress, or the user is at least a teacher and has a valid connection
+            // with this user...
+            if (user.getId().equals(userOfInterestSummary.getId())
+                    || isUserTeacherOrAbove(userManager, user) && associationManager.hasPermission(user, userOfInterestSummary)
```

Comment:
Is this to prevent tutors from seeing My Progress pages? It's not obvious that this is what it is doing, nor why?

Reply:
There are a few other places where this pattern occurs; it would be nice to wrap this all up in the AssociationManager as a sensibly-named method, to check role and permission together.

However, it raises another issue; currently if your account is demoted from Teacher to Student, you can still access all the connected students' data, because we only check for the existence of a connection. This will break that. That's probably for the best, but then, maybe the existing check should already verify Tutor or above, and we have a new method `hasTeacherPermission` that's slightly stricter?

Reply:
`isUserXOrAbove` and variants aren't available in `UserAssociationManager` since it isn't a façade, so I had to implement this in a slightly messy way, but it works as intended for our current role structure.